### PR TITLE
#4 split codebase to separate projects

### LINKED ISSUE #4: Split codebase to separate projects
Split to three projects?
- [x] C# (Core)
- [x] Grasshopper
- [x] Dynamo

Instead of three branches on github.
Releases will then follow with 3 different packages for the languages/UIs.

### Before merge
Ensure the following has been included in the new c# project:
- [x] All `#region dynamo`
- [x] All `[IsVisibleInDynamoLibrary(true)]` methods

Remove from repo:
- [x] `convertLibrary` scripts (not needed any longer)
- [x] `src/` files (these are the ones split to separate C# projects)

### Found problems to be aware of
- Enums cannot be partial and therefore can't be set to `[IsVisibleInDynamoLibrary(false)]`
- Methods cannot be partial and therefore methods of hidden Core classes cannot be set to `[IsVisibleInDynamoLibrary(true)]`.
- Properties cannot be partial...

### Classes not yet fixed. 
`[IsVisibleInDynamoLibrary(true)]` problems  
- ~Bars/Byckling/BucklingLength **visible**~
- ~Bars/Byckling/BucklingData **visible**~
- ~Bars/Connectivity **Not fixed**~
- ~Bars/Eccentricity **visible**~
- ~Materials/Material **Not fixed**~
- ~Materials/MaterialDatabase **Not fixed**~
- ~Model/Model **Missing visible properties**~
- ~ModellingTools/FictitiousShell **Missing method**~
- ~Releases/DetachType **Not hidden enum**~
- ~Shells/ShellEccentricity **Missing method**~
- ~Shells/ShellEdgeConnection **Missing 4 methods**~
- ~Shells/ShellOrthotropy **Missing 2 methods**~

Methods named somethoing like `GetXXXXX` or `SetXXXXX` or `XXXXDefine` etc, that should be visible in dynamo on a hidden class has now a Core method named `XXXXX` while Dynamo have a wrapper method named `GetXXXXX` as previously

### Will not fix
- RevitTools/ProjectSettings **removed**

### PR DESCRIPTION
**Pull request by xRadne**
Fix #4 split codebase to separate projects.

From issue:  
### Before merge
Ensure the following has been included in the new c# project:
- [x] All `#region dynamo`
- [x] All `[IsVisibleInDynamoLibrary(true)]` methods

Remove from repo:
- [x] `convertLibrary` scripts (not needed any longer)
- [x] `src/` files (these are the ones split to separate C# projects and thus not needed any longer)

Update:
- [x] Repository `README.md`

### Found problems to be aware of
- Enums cannot be partial and therefore can't be set to `[IsVisibleInDynamoLibrary(false)]`. These must be put in a separate file in the core project and **copied** (instead of linked) in dynamo project. 
- Methods cannot be partial and therefore methods of hidden Core classes cannot be set to `[IsVisibleInDynamoLibrary(true)]` and must have a **wrapper method** in dynamo project.
- Properties cannot be partial and must have a **wrapper method** to be exposed in dynamo.

Methods named somethoing like `GetXXXXX` or `SetXXXXX` or `XXXXDefine` etc, that should be visible in dynamo on a hidden class has now a Core method named `XXXXX` while Dynamo have a wrapper method named `GetXXXXX` as previously

### Not fixed
- [x] RevitTools/ProjectSettings **removed**

### CODE REVIEW

**Comment by andosca** [comment]
### Review checks:
#### Dynamo

- [ ] Compile
- [ ] Run dynamo_debug.dyn test
- [ ] Run dynamo_analysis.dyn test

#### Grasshopper

- [ ] Compile
- [ ] Run gh_debug.gh test
- [ ] Run gh_analysis.gh test

**Review by andosca — COMMENTED** [review]
Ok.

File: FemDesign.Dynamo/PackageData/pkg.json [hunk 1]
```diff
@@ -15,6 +15,6 @@
   "repository_url": "",
   "contains_binaries": true,
   "node_libraries": [
-    "FemDesign.Dynamo, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null"
+    "FemDesign.Dynamo, Version=1.3.0.0, Culture=neutral, PublicKeyToken=null"
```
Comment (andosca) [inline FemDesign.Dynamo/PackageData/pkg.json:18]: Latest version was actually 20.0.0.

Next version should be 20.1.0 or 20.0.1 depending on the amount of new features added.

**Comment by andosca** [comment]
# Review notes
## Compile
AssemblyInfo and Resources are missing. This creates some friction in setting up the dev. environment.

File: FemDesign.Dynamo/PackageData/pkg.json (hunk 1 above)
Comment (xRadne) [reply FemDesign.Dynamo/PackageData/pkg.json:18]: I suggest we increase this when it is time for a new release, which could be in a separate PR?

**Comment by xRadne** [comment]
@andosca It would be great to have this merged before I make my presentation of the API. Could we get this closed tomorrow?

- [ ] What do you mean with this? 
> Grasshopper directories directly in project folder.  

That the grasshopper components are placed in the root of `FemDesign.Grasshopper`? I think this is good since that follows the same structure as with `FemDesign.Core`. 
(The dynamo .csproj however uses both the files in Core and Dynamo and these have been placed in `FemDesign.Dynamo/Dynamo` to be separate from `FemDesign.Dynamo/Core`)

- [ ] The Buckling length issue #41. This seems unrelated to the split of the repo. Correct?

**Comment by andosca** [comment]
> Grasshopper directories directly in project folder.

I agree with you - let's leave it.

> The Buckling length issue #41. This seems unrelated to the split of the repo. Correct?
I'll move it out of this PR.

If the dynamo_deconstruct (a.k.a dynamo_debug) passes. Should be ok to merge.

**Pull request merged**